drivers/usb/udc/udc_stm32.c

```diff
@@ -991,23 +995,12 @@ static void priv_pcd_prepare(const struct device *dev)
 
 	memset(&priv->pcd, 0, sizeof(priv->pcd));
 
-	/* Default values */
 	priv->pcd.Init.dev_endpoints = cfg->num_endpoints;
 	priv->pcd.Init.ep0_mps = cfg->ep0_mps;
-	priv->pcd.Init.speed = UTIL_CAT(UDC_STM32_, DT_INST_STRING_UPPER_TOKEN(0, maximum_speed));
-
 	/* Per controller/Phy values */
-#if defined(USB)
-	priv->pcd.Instance = USB;
-#elif defined(USB_DRD_FS)
-	priv->pcd.Instance = USB_DRD_FS;
-#elif defined(USB_OTG_FS) || defined(USB_OTG_HS)
-#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs)
-	priv->pcd.Instance = USB_OTG_HS;
-#else
-	priv->pcd.Instance = USB_OTG_FS;
-#endif
-#endif /* USB */
+	priv->pcd.Init.speed = UTIL_CAT(UDC_STM32_, DT_INST_STRING_UPPER_TOKEN(0, maximum_speed));
+	priv->pcd.Instance = (PCD_TypeDef *)DT_INST_REG_ADDR(0);
+	priv->pcd.Init.phy_itface = DT_INST_STRING_UPPER_TOKEN(0, physel);
 
 #if USB_OTG_HS_EMB_PHY
```

Comment:
It seems you forgot to remove the lines below in your last P-R update (commit "drivers: usb: udc_stm32: refactor USB phy_itface assignment").